tests/validation/tests/v3_api/test_node.py

```diff
@@ -1,54 +1,1072 @@
+
+    test_label = random_name()
+    label_value = random_name()
+    # create a node template with a label
+    node_template_new, do_cloud_credential = \
+        create_node_template_label(client, test_label, label_value)
+    # Add a node in cluster
+    cluster, node_pools = add_node_cluster(node_template_new, cluster)
+    nodes = client.list_node(clusterId=cluster.id).data
+
+    # validate labels on nodes
+    for node in nodes:
+        if node.nodeName not in existing_labels.keys():
+            # check if label is set on node
+            validate_label_set_on_node(client, node, test_label, label_value)
+        else:
+            # check if the labels on the existing nodes are intact
+            existing_labels[node.nodeName] == node.labels.data_dict()
+
+
+@pytest.mark.run(after='test_node_label_node_template_edit')
+def test_node_label_node_template_edit():
+    client = cluster_detail["client"]
+    cluster = cluster_node_template["cluster"]
+    node_template = cluster_node_template["node_template"]
+    do_cloud_credential = cluster_node_template["do_cloud_credential"]
+    test_label = cluster_node_template["test_label"]
+    create_kubeconfig(cluster)
+    nodes = client.list_node(clusterId=cluster.id).data
+
+    existing_labels = {}
+    for node in nodes:
+        existing_labels[node.nodeName] = {}
+        existing_labels[node.nodeName] = node.labels.data_dict()
+
+    template_label = node_template.labels.data_dict()
+    assert test_label in template_label, \
+        "Label is NOT available on the node template"
+
+    new_value = random_name()
+    template_label[test_label] = new_value
+    node_template_new = client.update(node_template, labels=template_label,
+                                      cloudCredentialId=do_cloud_credential.id,
+                                      digitaloceanConfig=
+                                      {"region": "nyc3",
+                                       "size": "2gb",
+                                       "image": "ubuntu-16-04-x64"})
+    node_template_new = client.wait_success(node_template_new)
+    remove_resource(node_template_new)
+    assert test_label in node_template_new["labels"], \
+        "Label is not set on node template"
+    assert node_template_new["labels"][test_label] == new_value
+
+    # Add a node in cluster
+    cluster, node_pools = add_node_cluster(node_template_new, cluster)
+
+    nodes = client.list_node(clusterId=cluster.id).data
+    """check original label on the first node,
+    and the new label on the added node"""
+
+    # validate labels on nodes
+    for node in nodes:
+        if node.nodeName not in existing_labels.keys():
+            # check if label is set on node
+            validate_label_set_on_node(client, node, test_label, new_value)
+        else:
+            # check if the labels on the existing nodes are intact
+            existing_labels[node.nodeName] == node.labels.data_dict()
```

Comment:
what if a key is missing or a value is . different? Does it give true or false? Check

Reply:
Missing key, different keys, different values of keys, additional key/value pairs -- all these are checked in this type of comparison